LeaderboardBackend.Test/Categories.cs

```diff
@@ -97,6 +101,209 @@ public static async Task CreateCategory_GetCategory_OK()
             $"/api/category/{createdCategory?.Id}", new() { }
         );
 
-        retrievedCategory.Should().BeEquivalentTo(request, opts => opts.Excluding(c => c.LeaderboardId));
+        retrievedCategory.Should().BeEquivalentTo(request);
+    }
+
+    [Test]
+    public static async Task CreateCategory_Unauthenticated()
+    {
+        Instant now = Instant.FromUnixTimeSeconds(1);
+        _clock.Reset(now);
+
+        CreateCategoryRequest request = new()
+        {
+            Name = "Unauthenticated",
+            Slug = "unauthn",
+            Info = "",
+            SortDirection = SortDirection.Ascending,
+            Type = RunType.Time
+        };
+
+        await FluentActions.Awaiting(() => _apiClient.Post<CategoryViewModel>(
+            $"/leaderboard/{_createdLeaderboard.Id}/categories/create",
+            new()
+            {
+                Body = request,
+            }
+        )).Should().ThrowAsync<RequestFailureException>().Where(e => e.Response.StatusCode == HttpStatusCode.Unauthorized);
+    }
+
+    [TestCase(UserRole.Banned)]
+    [TestCase(UserRole.Confirmed)]
+    [TestCase(UserRole.Registered)]
+    public static async Task CreateCategory_BadRole(UserRole role)
+    {
+        IServiceScope scope = _factory.Services.CreateScope();
+        IUserService userService = scope.ServiceProvider.GetRequiredService<IUserService>();
+        ApplicationContext context = scope.ServiceProvider.GetRequiredService<ApplicationContext>();
+
+        string email = $"testuser.updatelb.{role}@example.com";
```

Comment:
Should be `testuser.updatecat.{role}@example.com` I suppose.

Reply:
Oh oops